### PR TITLE
Improve iscsi validation

### PR DESCRIPTION
**Pull request by sonicaj**
This commit improves validation for querying portals and initiators before trying to insert them as a foreign key in database.
Ticket: #70581

### CODE REVIEW

File: src/middlewared/middlewared/plugins/iscsi.py [hunk 1]
```diff
@@ -1008,14 +1008,51 @@ class Config:
...
+                    f'{schema_name}.groups.{i}.portal',
+                    f'{group["portal"]} Portal does not exist in database'
+                )
             else:
                 portals.append(group['portal'])
 
+            if group['initiator'] and group['initiator'] not in db_initiators:
+                verrors.add(
+                    f'{schema_name}.groups.{i}.initiator',
+                    f'{group["initiator"]} Initiator does not exist in database'
```
Comment (wblock) [inline src/middlewared/middlewared/plugins/iscsi.py]: s/does not exist/not found/

File: src/middlewared/middlewared/plugins/iscsi.py [hunk 2]
```diff
@@ -1008,14 +1008,51 @@ class Config:
...
         #    verrors.add(f'{schema_name}.groups', 'At least one group is required')
 
+        db_portals = list(
+            map(
+                lambda v: v['id'],
+                await self.middleware.call('datastore.query', 'services.iSCSITargetPortal', [
+                    [
+                        'OR', [
+                            ('id', '=', portal_id) for portal_id in map(lambda v: v['portal'], data['groups'])
+                        ]
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/iscsi.py]: Isnt OR somewhat overkill? Wouldnt ('id', 'in', portal_ids) do just the same?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/iscsi.py]: It would, thank you :)

**Comment by sonicaj** [comment]
backport

**Pull request merged**